yt_dlp/extractor/youtube.py

```diff
@@ -2487,6 +2500,11 @@ def _extract_player_responses(self, clients, video_id, webpage, master_ytcfg, pl
                 if client in INNERTUBE_CLIENTS and client not in original_clients:
                     clients.append(client)
 
+                # Try web creator (studio) client if session cookies provided (bypass for "AGE_VERIFICATION_REQUIRED")
+                if self._generate_sapisidhash_header() is not None and not creator_client_appended:
+                    clients.append("web_creator")
+                    creator_client_appended = True
```

Comment:
github wont let me comment on the above line. The suggestion is meant to replace lines 2499-2506

```py
                client = f'{client}_creator' if self._generate_sapisidhash_header() else f'{client}_agegate'
                if client in INNERTUBE_CLIENTS and client not in original_clients:
                    clients.append(client)
```

Isn't this enough? This will only fallback to creator if cookies are given and `web` throws an age-gate error. Also, in that case, the normal age-gate bypass is not attempted (should it be?). If we can add `android_creator`/`ios_creator`, the fallbacks will automatically extend to them